lib/cloud/clients.go

```diff
@@ -630,6 +645,21 @@ func (c *cloudClients) initGCPGKEClient(ctx context.Context) (gcp.GKEClient, err
 	return gcpGKE, nil
 }
 
+func (c *cloudClients) initGCPInstancesClient(ctx context.Context) (gcp.InstancesClient, error) {
```

Comment:
This function is the 4th copy of the same code. Can we, at some point, either simplify the code or extract the logic? Right now even the comment is the same in all of them.